endpoints/resources/src/main.cpp

```diff
@@ -221,49 +222,133 @@ namespace
 
 	IRODS_HTTP_API_ENDPOINT_OPERATION_SIGNATURE(op_modify)
 	{
-#if 0
         auto result = irods::http::resolve_client_identity(_req);
         if (result.response) {
             return _sess_ptr->send(std::move(*result.response));
         }
 
         const auto client_info = result.client_info;
 
-        irods::http::globals::background_task([fn = __func__, client_info, _sess_ptr, _req = std::move(_req), _args = std::move(_args)] {
-            logging::info("{}: client_info.username = [{}]", fn, client_info.username);
+		irods::http::globals::background_task([fn = __func__,
+		                                       client_info,
+		                                       _sess_ptr,
+		                                       _req = std::move(_req),
+		                                       _args = std::move(_args)] {
+			logging::info("{}: client_info.username = [{}]", fn, client_info.username);
+
+			http::response<http::string_body> res{http::status::ok, _req.version()};
+			res.set(http::field::server, irods::http::version::server_name);
+			res.set(http::field::content_type, "application/json");
+			res.keep_alive(_req.keep_alive());
+
+			try {
+				const auto name_iter = _args.find("name");
+				if (name_iter == std::end(_args)) {
+					logging::error("{}: Missing [name] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				const auto property_iter = _args.find("property");
+				if (property_iter == std::end(_args)) {
+					logging::error("{}: Missing [property] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				const auto value_iter = _args.find("value");
+				if (value_iter == std::end(_args)) {
+					logging::error("{}: Missing [value] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				auto conn = irods::get_connection(client_info.username);
+
+				if (property_iter->second == "name") {
+					// TODO(???): Remove this once the resource administration library grows support
+					// for renaming resources.
+					//
+					// The following is equivalent to what the resource administration library would
+					// provide.
+					GeneralAdminInput input{};
+					input.arg0 = "modify";
+					input.arg1 = "resource";
+					input.arg2 = name_iter->second.c_str();
+					input.arg3 = "name";
+					input.arg4 = value_iter->second.c_str();
+
+					if (const auto ec = rcGeneralAdmin(static_cast<RcComm*>(conn), &input); ec < 0) {
+						constexpr const char* msg = "Could not change name from [{}] to [{}] for resource [{}]";
+						// NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+						THROW(ec, fmt::format(msg, input.arg3, input.arg4, input.arg2));
```

Comment:
Feels like this message needs to be reworded or something. Currently, I think it will print something like this if we are attempting to rename a resource from resc1 to resc2:

"Could not change name from [name] to [resc2] for resource [resc1]."

An initial suggestion:

"Could not change property [name] to [resc2] for resource [resc1]."

...or something like that.

Reply:
Good catch. Will need to adjust it in irods/irods as well.

Reply:
Done.

Created irods/irods#7877 too.